### PR TITLE
AWS Upload

### PR DESCRIPTION
**Pull request by Diniden**
feature: When AWS_BUCKET and AWS credentials are present in the environment, the process during auto release will tar the newly made distribution

### CODE REVIEW

**Review by tarwich — COMMENTED** [review]
Can you remove your .DS_Store files and add them to .gitignore?

File: actions/aws-upload.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+ *
+ * This is an example to use this method straight from the command line (fish shell):
+ *
+ * env
+ *  TEST=true
+ *  AWS_ACCESS_KEY_ID=<insert your AWS access key here>
+ *  AWS_SECRET_ACCESS_KEY=<insert your AWS secret access key here>
+ *  AWS_PUBLIC=<set this to true to make the uploaded file public>
+ *  AWS_BUCKET=<set this to bucketName/path/to/file/filename.tar.gz
+ *  AWS_UPLOAD=<set this to the directory you want uploaded>
```
Comment (tarwich) [inline actions/aws-upload.js]: Is this a source directory? If so, then perhaps we could rename it to be a little clearer.

File: actions/aws-upload.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  if (awsFileName) {
+    // Add the file name as a part of the pathing
+    bucketPathing.push(awsFileName);
+  }
+
+  const myKey = bucketPathing.slice(1).join('/');
+
+  // Validate all inputs for the process
+  if (!bucketPath || !bucketPathing || !pathToUpload) {
+    reject(new Error(`All needed inputs for the AWS upload were not provided or invalid. bucket: ${myBucket}, bucketPath: ${myKey}, toUpload: ${pathToUpload}`));
```
Comment (tarwich) [inline actions/aws-upload.js]: What is `reject` at this point? I don't see it. I think you mean `throw`

**Review by tarwich — COMMENTED** [review]
Is test.tar.gz necessary?

File: actions/aws-upload.js (hunk 2 above)
Comment (Diniden) [reply actions/aws-upload.js]: Ah missed that one, converted from Promises 👍 will dew

**Review by tarwich — APPROVED** [review]

**Pull request merged**